### ISSUE TITLE
Add the ability to define searchable fields

### ISSUE THREAD

**Issue by jrgns** [EagerELK/ditty#1]
The `Component` helper currently allows a developer to specify [filter fields](https://github.com/EagerELK/ditty/blob/master/lib/ditty/helpers/component.rb#L42) for a model that allows you to filter list results according to that field. As an example https://ditty.io/products?type=online will only return products where the `type` field equals `online`

Extend the `Component` helper to allow a developer to specify search fields in a controller:

```ruby
module Ditty
  class Product < Ditty::Component
    SEARCHABLE = [ :description, :name ]
  end
end
```

Where each specified field will be included in the DB search. In the example above, this URL will search for the term `free` in the `description` and `name` fields: field: https://ditty.io/products?q=free

**Comment by tegon** [comment]
Hi! Can I tackle this one?

**Comment by jrgns** [comment]
@tegon Please do!

**Comment by jrgns** [comment]
@tegon Still keen to do this?

**Comment by tegon** [comment]
@jrgns I haven't started working on this yet, I have some other tasks to finish before.
If you want to do it or give it to someone else, feel free to do it. There's no problem for me.

**Comment by tegon** [comment]
@jrgns I've implemented this in #4. Please let me know what you think of the solution.

**Comment by jrgns** [comment]
Thanx @tegon I've added you to the Contributors file.
